collect_app/src/main/java/org/odk/collect/android/external/InstanceProvider.java

```diff
@@ -132,6 +133,10 @@ public Uri insert(@NonNull Uri uri, ContentValues initialValues) {
             throw new IllegalArgumentException("Unknown URI " + uri);
         }
 
+        if (initialValues.containsKey(DatabaseInstanceColumns.SUBMISSION_URI)) {
```

Comment:
This solution is not sufficient to me. It is still possible to update `DatabaseFormColumns.SUBMISSION_URI` (the one from the `forms.db`) in ta similar way suing content providers and then every new saved form will have that new submission uri. I think we should block both ways.